jobmanager/WorkerThread.cpp

```diff
@@ -2,14 +2,35 @@
 #include "concurrency/ThreadUtils.h"
 #include "jobmanager/JobManager.h"
 
+namespace
+{
+
+thread_local jobmanager::WorkerThread* workerThreadHandler = nullptr;
+
+} // namespace
+
 namespace jobmanager
 {
 
 WorkerThread::WorkerThread(jobmanager::JobManager& jobManager)
     : _running(true),
       _jobManager(jobManager),
+      _backgroundJobCount(0),
       _thread([this] { this->run(); })
 {
+    _backgroundJobs.reserve(512);
+    std::fill(_backgroundJobs.begin(), _backgroundJobs.end(), nullptr);
```

Comment:
This is doing nothing. Reserve only reserve space internally on memory managed by std::vector. But it change the size (the elements that actually being used by vector). Then  in this point `backgroundJobs.begin()` is equal to `_backgroundJobs.end()`